airflow/utils/db.py

```diff
@@ -627,27 +627,32 @@ def check_migrations(timeout):
     :param timeout: Timeout for the migration in seconds
     :return: None
     """
-    from alembic.runtime.migration import MigrationContext
+    from alembic.runtime.environment import EnvironmentContext
     from alembic.script import ScriptDirectory
 
     config = _get_alembic_config()
     script_ = ScriptDirectory.from_config(config)
-    with settings.engine.connect() as connection:
-        context = MigrationContext.configure(connection)
-        ticker = 0
-        while True:
-            source_heads = set(script_.get_heads())
-            db_heads = set(context.get_current_heads())
-            if source_heads == db_heads:
-                break
-            if ticker >= timeout:
-                raise TimeoutError(
-                    f"There are still unapplied migrations after {ticker} seconds. "
-                    f"Migration Head(s) in DB: {db_heads} | Migration Head(s) in Source Code: {source_heads}"
-                )
-            ticker += 1
-            time.sleep(1)
-            log.info('Waiting for migrations... %s second(s)', ticker)
+    with EnvironmentContext(
+        config,
+        script_,
+    ) as env:
+        with settings.engine.connect() as connection:
```

Comment:
```suggestion
    with EnvironmentContext(
        config,
        script_,
    ) as env, settings.engine.connect() as connection:
```

Is this syntax available yet?

Reply:
Oh yes https://docs.python.org/3.6/reference/compound_stmts.html#the-with-statement

Reply:
Done.